sound/soc/intel/boards/sof_es8336.c

```diff
@@ -174,18 +174,18 @@ static int sof_es8336_hw_params(struct snd_pcm_substream *substream,
 		gpiod_set_value_cansleep(ctx->gpio_pa, true);
 	}
 
-	ret = snd_soc_dai_set_sysclk(codec_dai, 1,
-				     sysclk, SND_SOC_CLOCK_OUT);
+	ret = snd_soc_dai_set_sysclk(codec_dai, 1, sysclk, SND_SOC_CLOCK_OUT);
 	if (ret < 0) {
 		dev_err(rtd->dev, "%s, Failed to set ES8336 SYSCLK: %d\n",
 			__func__, ret);
 		return ret;
 	}
+
 	ret = snd_soc_dai_set_tdm_slot(codec_dai, 0x3, 0x3, 2, 24);
-	if (ret < 0)
-		return ret;
+	if (ret == -ENOTSUPP)
+		ret = 0;
```

Comment:
@hli25 can you explain this line? This would mean it's not necessary to set the TDM slots?
Can we remove lines 184..186 then?

Reply:
thanks for your comment.
The PR  was updated to remove lines 184...186, and verified it on my glk device.

previously i thought tx_mask and rx_mask should be assigned new value even if there was no set_tdm_slot() callback.